lib/new_relic/agent/attribute_processing.rb

```diff
@@ -57,6 +59,101 @@ def flatten_and_coerce_array(array, prefix, result, &blk)
           end
         end
       end
+
+      def formulate_regexp_union(option)
```

Comment:
TODO: there isn't any cross dependency on filtering and pre-filtering, so move the pre-filtering stuff out into its own class. Perhaps `pre_attribute_processing.rb`

Reply:
addressed with [2fa3e66](https://github.com/newrelic/newrelic-ruby-agent/pull/2177/commits/2fa3e6664096ea033fab3d177513afd49b1185b6)